structured/structured.go

```diff
@@ -140,55 +145,64 @@ func (desc *TableDescriptor) Validate() error {
 
 	// TODO(pmattis): Check that the indexes are unique. That is, no 2 indexes
 	// should contain identical sets of columns.
-
-	if len(desc.Indexes) == 0 {
-		return fmt.Errorf("table must contain at least 1 index")
+	if desc.PrimaryIndex.Name == "" {
+		return fmt.Errorf("table must contain a primary index")
 	}
 
 	indexNames := map[string]struct{}{}
 	indexIDs := map[uint32]string{}
+	if err := validateIndex(desc.PrimaryIndex, desc.NextIndexID, &indexNames, &indexIDs, columnNames); err != nil {
```

Comment:
append thing here too. perhaps a convenience method on TableDescriptor that returns all the indexes would be helpful.